### DISCUSSION TITLE
[BUG] "Get-PnPListItem -List mylist" throws error for some lists.

### DISCUSSION THREAD

**Discussion by Larryblanco-git** [pnp/powershell#3932] [General]
### Reporting an Issue or Missing Feature
Get-PnPListItem
This command is failing to list items in some lists.
 
### Expected behavior 

Return a list of items in list
Get-PnPListItem -List "service Centers"

Id    Title                                              GUID
--    -----                                              ----
1     Y1                                         2d691f57-93f7-4978-b3d9-0fb27923e6ed
2     Y3                                         3124b46a-8298-49de-9ebc-0ea29ac2f373
3     Y5                                         66a26e52-285e-4fe6-9fe4-e1259b5e146f
4     Y4                                         f015218c-ee4e-4423-8ef8-ddda0dd5d4e9

### Actual behavior 
Get-PnPListItem -List cameras

Get-PnPListItem: Cannot invoke method or retrieve property from null object. Object returned by the following call stack is null. "GetList
"
### Steps to reproduce behavior
Please include complete script or code samples in-line or linked from [gists](https://gist.github.com/)

Connect-PnPOnline -Url $SiteURL -Credentials $CredentialSPOA
Get-PnPListItem -List cameras

### What is the version of the Cmdlet module you are running?
2.4.0

### Which operating system/environment are you running PnP PowerShell on?
- [X] Windows
- [x] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by astapelfeld** [comment]
Same issue on Linux
Module Version 2.4.0
PowerShell 7.4.1

**Comment by Larryblanco-git** [comment]
This is the output of the exception in a try-catch:

PSMessageDetails      : 
Exception             : System.Management.Automation.PSInvalidOperationException: Cannot invoke method or retrieve property from null object. Object returned by the following call stack is null. "File "
                           at PnP.PowerShell.Commands.Base.PnPConnectedCmdlet.ProcessRecord() in c:\build\src\Commands\Base\PnPConnectedCmdlet.cs:line 101
                           at PnP.PowerShell.Commands.PnPSharePointCmdlet.ProcessRecord() in c:\build\src\Commands\Base\PnPSharePointCmdlet.cs:line 131
                           at System.Management.Automation.CommandProcessor.ProcessRecord()
TargetObject          : 
CategoryInfo          : InvalidOperation: (:) [Set-PnPListItem], PSInvalidOperationException
FullyQualifiedErrorId : InvalidOperation,PnP.PowerShell.Commands.Lists.SetListItem
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at <ScriptBlock>, C:\Files\Link.ps1: line 152 at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {} 

Anyone have an idea of what may be wrong?

**Comment by markgort86** [comment]
In the last comment it looks like you're running the cmdlet Set-PnPListItem instead of Get-PnPListItem (where this discussion should be about). Is this correct?

**Reply by Larryblanco-git** [reply]
You are correct, and I am now I'm scratching my head.  I'm certain I was having issues with 'Get-PnPListItem'.  It seems now I can pull the list but cannot update it.  Below is a snippet of the script:



```powershell
$BogusQueryString = "?Query=" + $(get-date -UFormat %s)
$ListItems = Get-PnPListItem -List $ListName -PageSize 2000

#Update image url with bogus query string to have the image reload -circumvent image caching on SharePoint
ForEach ($Item in $ListItems)
{
	$OldHttpURL = $Item.FieldValues[$ColumnName].URL.ToLower()
	$NewHttpURL = (($OldHttpURL -split "\?Query=", 2)[0]) + $BogusQueryString
	$Hyperlink = New-Object Microsoft.SharePoint.Client.FieldUrlValue
	$Hyperlink.Url = $NewHttpURL
	Try
	{
		Set-PnPListItem -List $ListName -Identity $Item.Id -Values @{ $ColumnName = [Microsoft.SharePoint.Client.FieldUrlValue]$Hyperlink } | Out-Null
	}
	Catch
	{
		$_ | FL * -force
	}
}
``` 

the variables look like this:

OldHttpURL: 
https://SOMETENANT.sharepoint.com/cameras/cameraimages/site/y1/chimneyf1.jpg?query=1717617785

NewHttpURL:
https://SOMETENANT.sharepoint.com/cameras/cameraimages/site/y1/chimneyf1.jpg?Query=1717618528

Hyperlink.URL:
https://SOMETENANT.sharepoint.com/cameras/cameraimages/site/y1/chimneyf1.jpg?Query=1717618528


Larry

**Comment by AndersRask** [comment]
Old and new URL are different, so what is it that is not working?

I tried updating a hyperlink using set-pnplistitem without issues.
